service_crategen/src/commands/generate/codegen/rest_json.rs

```diff
@@ -113,11 +113,15 @@ impl GenerateProtocol for RestJsonGenerator {
         Ok(())
     }
 
-    fn generate_prelude(&self, writer: &mut FileWriter, _: &Service) -> IoResult {
-        writeln!(writer,
-                 "use serde_json;
-        use rusoto_core::param::{{Params, ServiceParams}};
-        use rusoto_core::signature::SignedRequest;
+    fn generate_prelude(&self, writer: &mut FileWriter, service: &Service) -> IoResult {
+        writeln!(writer, "use serde_json;")?;
+
+        // avoid unused imports when building services that don't use params
+        if service.service_type_name() != "Batch" {
```

Comment:
While I'd like to not have to special case things, I'd prefer we only include code we need rather than allowing unused imports.  For better or worse we already have some service-specific bits in various generators ([one](https://github.com/rusoto/rusoto/blob/master/service_crategen/src/commands/generate/codegen/rest_xml.rs#L296), [two](https://github.com/rusoto/rusoto/blob/master/service_crategen/src/commands/generate/codegen/query.rs#L254) ).

I may be overly sensitive to this due to the project's history with long compile times, but I prefer not bringing in unused code. 😄